tests/lib/prepare.sh

```diff
@@ -524,7 +524,20 @@ uc20_build_initramfs_kernel_snap() {
     # TODO: install the linux-firmware as the current version of
     # ubuntu-core-initramfs does not depend on it, but nonetheless requires it
     # to build the initrd
-    apt install ubuntu-core-initramfs linux-firmware -y
+    apt install linux-firmware devscripts equivs -y
+
+    git clone -b core20 https://github.com/snapcore/core-initrd.git
```

Comment:
building core-initrd from source on my machine takes quite a while due to having to build systemd from source, so it would be nice to not have to do that every uc20 spread run.

Additionally, it's probably not super clear in the code, but at the time we first set all this script infrastructure up, we wanted to make sure that we are mainly testing whatever core-initrd version was released with an actual kernel snap, and then just inject our own snap-bootstrap inside that to test the new version of snap-bootstrap, because we were concerned about having bugs show up on the main branch of core-initrd and randomly break our CI frequently. That's probably less of a concern these days now that UC20 is released, but still I think we should be cautious with adding too many dependencies in our tests which follow a master branch because we do suffer from having our CI break when others' break things on their own development branch and having to go debug their issues for them. 

I think what I proposed to Dimitri at some point was to have a separate (probably nested) spread test for building core-initrd from git, and all the rest of the spread tests would do like we do today and unpack what is in the released kernel snap and inject snap-bootstrap from the branch there. So to accommodate that, you could have a flag/option for this uc20_build_initramfs_kernel_snap function which controls whether to inject snap-bootstrap into an already built initramfs or whether this should build the initramfs deb from scratch like you are doing in this pull request.